sim/midas/src/main/scala/midas/passes/fame/PromotePassthroughConnections.scala

```diff
@@ -0,0 +1,136 @@
+// See LICENSE for license details.
+
+package midas.passes.fame
+
+import firrtl._
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.traversals.Foreachers._
+import firrtl.annotations._
+import firrtl.annotations.TargetToken.{OfModule, Instance, Field}
+import firrtl.stage.Forms
+import firrtl.options.Dependency
+import firrtl.transforms.{CheckCombLoops, LogicNode}
+
+import scala.collection.mutable
+
+/** Removes all non-passthrough expressions in a circuit so that we can use
+  *  CheckCombLoops to return only passthrough connectivity (this represents
+  *  subset of of all combinationally connected paths).
+  *
+  * @param state the CircuitState to simplify
+  */
+object RemoveNonWirePrimitives {
+  def apply(state: CircuitState): CircuitState = {
```

Comment:
This should probably work on a `Circuit`, rather than a `CircuitState`, since it's not part of a coherent transform.